docs/docs/sql/string.md

```diff
@@ -186,3 +199,273 @@ SELECT 'h%wkeye' LIKE 'h#%%' ESCAPE '#'        true
 SELECT 'h%wkeye' NOT LIKE 'h#%%' ESCAPE '#'    false
 SELECT 'h%awkeye' LIKE 'h#%a%k%e' ESCAPE '#'   true
 ```
+
+## POSIX regular expressions
+
+POSIX regular expressions provide a more powerful means for pattern
+matching than the `LIKE` and `SIMILAR TO` operators.  Many Unix tools
+such as `egrep`, `sed`, or `awk` use a pattern matching language that
+is similar to the one described here.
+
+Currently our compiler does not support `SIMILAR TO` regular
+expressions.
+
+A regular expression is a character sequence that is an abbreviated
+definition of a set of strings (a regular set). A string is said to
+match a regular expression if it is a member of the regular set
+described by the regular expression. As with `LIKE`, pattern
+characters match string characters exactly unless they are special
+characters in the regular expression language — but regular
+expressions use different special characters than `LIKE` does. Unlike
+`LIKE` patterns, a regular expression is allowed to match anywhere
+within a string, unless the regular expression is explicitly anchored
+to the beginning or end of the string.
+
+The description below is from the Postgres documentation, where credit
+is given to Henry Spencer.
```

Comment:
We should be following the terms of the postgres license here if the description is borrowed verbatim. Maybe we should just link to it?

Reply:
I don't know if the postgres license applies to documentation too.
We get documentation left and right from Postgres, Calcite, Spark, whatever has useful stuff.

Reply:
> I don't know if the postgres license applies to documentation too. We get documentation left and right from Postgres, Calcite, Spark, whatever has useful stuff.

Most free software doesn't have a different license for the documentation.

The Postgres license is very liberal (see https://www.postgresql.org/about/licence/). I think that it would be good practice to include it alongside our other licenses and mention that we've used some of the Postgres documentation as part of ours.